### PR TITLE
fix: keyboard nav & focus in Explore control panel

### LINKED ISSUE #14574: Keyboard navigation in Explore control panel
Keyboard navigation in the control panel of the Explore view is difficult.

### Expected results

You should be able to move focus between adjacent controls in the control panel with a single Tab key press
and visually distinguish what element has focus. You should be able to interact with controls the keyboard
(Enter or space bar for button-like things).

### Actual results

Several tab key presses are required to move between controls, and focus is not always visible.

#### How to reproduce the bug

1. Go to a chart in Explore view (from Welcome or Chart List).
2. Click on a control in the control panel (e.g. 'TIME COLUMN') to focus that control.
3. Press the tab key
4. The next control (e.g. 'TIME GRAIN') should be focused, but focus seems to disappear. (in this case,
   a second tab press will move focus to the expected place).

### Environment

(please complete the following information):

- superset version: `master` branch, but has been present since at least `0.38`
- python version: `3.7.10`
- node.js version: `v14.6.1`
- npm version: `7.12.1`

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [X] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X] I have reproduced the issue with at least the latest released version of superset.
- [X] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

This seems to mostly stem from `ControlHeader` being treated as a button (including `tabindex` and `role`) in 
order to emulate the browser's behavior when clicking on a form `label` for custom elements. For most
controls, including the HTML `for` attribute will result in the desired behavior (some components, like
`Checkbox`, will still need the click handler).

For components that correctly get focus (most notably here the `CollapsePanel` headers), there needs to be a
visual indicator, possibly the same as the hover style.

Some of the more complex components will be affected by the outcome of [SIP-62](https://github.com/apache/superset/issues/14114)

### PR DESCRIPTION
**Pull request by GregOnEvo**
### SUMMARY
Provides better keyboard navigation in control panel of Explore view. Allows `Label`-styled buttons and `Checkbox`es to focused & interacted with using keyboard. Emulates browser form label click behavior on `ControlHeader`s: focus, and for `Checkbox`es, select.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
n/a

### TEST PLAN
Navigate control panel using Tab/Shift-Tab; focus should visibly move in a sensible way.

### ADDITIONAL INFORMATION

- [X] Has associated issue: Fixes #14574
- [X] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/explore/components/controls/TextControl/index.tsx [hunk 1]
```diff
@@ -27,6 +27,7 @@ type InputValueType = string | number;
 
 export interface TextControlProps<T extends InputValueType = InputValueType> {
   label?: string;
+  name?: string;
```
Comment (geido) [inline superset-frontend/src/explore/components/controls/TextControl/index.tsx:30]: It seems that name should not be optional to promote the correct usage

**Comment by rusackas** [comment]
Not sure how this slipped under my radar for so long, but I appreciate this effort and will try to take a closer look!

**Comment by rusackas** [comment]
you can run `npm run lint-fix` in your console to fix the linting errors.

**Comment by GregOnEvo** [comment]
Thanks for taking a look, should be better now.

**Pull request closed without merging**